src/main/java/gov/nih/nci/bento_ri/model/PrivateESDataFetcher.java

```diff
@@ -431,6 +436,481 @@ private Map<String, List<Object>> idsLists() throws IOException {
+                        Object enrichedCpiDataArray = enrichedCPILookup.get(lookupKey);
+
+                        // Update the cpi_data field (or user-provided field) in the participant record with the enriched array
+                        participant.put(synonymsPropertyKey, enrichedCpiDataArray);
+                        // System.out.println("Updated participant " + participantId + " with enriched CPI data array");
+                    } else {
+                        // System.out.println("No enriched CPI data found for participant: " + participantId + ", study: " + studyId);
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("Error updating participant with enriched CPI data: " + e.getMessage(), e);
+            }
+        }
+
+        // System.out.println("Completed updating participant_list with enriched CPI data");
+    }
+
+    /**
+     * Puts CPI data into participants
+     */
+    private void insertCPIDataIntoParticipants(List<Map<String, Object>> participants) {
+        insertCPIDataIntoParticipants(participants, null);
+    }
+
+    /**
+     * Puts CPI data into participants
+     * @param participants List of participant objects
+     * @param synPropName The name of the synonyms property in the participant record
+     * @return void
+     * @throws Exception If an error occurs while putting CPI data into participants
+     */
+    private void insertCPIDataIntoParticipants(List<Map<String, Object>> participants, String synPropName) {
+        List<ParticipantRequest> cpiIDs = extractIDs(participants);
+
+        // Check if CPIFetcherService is properly injected
+        if (cpiFetcherService == null) {
+            logger.warn("CPIFetcherService is not properly injected. CPI integration will be skipped.");
+        } else {
+            try {
+                List<FormattedCPIResponse> cpiData = cpiFetcherService.fetchAssociatedParticipantIds(cpiIDs);
+                logger.info("CPI data received: " + cpiData.size() + " records");
+
+                // Print the first value as JSON
+                if (cpiData != null && !cpiData.isEmpty()) {
+                    // System.out.println("First CPI data value BEFORE enrichment: " + gson.toJson(cpi_data.get(0)));
+
+                    // Enrich CPI data with additional participant information
+                    enrichCPIDataWithParticipantInfo(cpiData);
+
+                    // Print the first enriched CPI data value
+                    // System.out.println("First enriched CPI data value AFTER enrichment: " + gson.toJson(cpi_data.get(0)));
+
+                    // Update the participant_list with the enriched CPI data
+                    if (synPropName == null) {
+                        updateParticipantListWithEnrichedCPIData(participants, cpiData);
+                    } else {
+                        updateParticipantListWithEnrichedCPIData(participants, cpiData, synPropName);
+                    }
```

Comment:
Hmm maybe this if statement isn't necessary, because `updateParticipantListWithEnrichedCPIData()` already handles a `synPropName = null`. Should I get rid of this and the two-parameter version of `updateParticipantListWithEnrichedCPIData()`?